posthog/caching/calculate_results.py

```diff
@@ -145,13 +146,23 @@ def calculate_for_query_based_insight(
     if isinstance(response, BaseModel):
         response = response.model_dump()
 
+    cache_key = response.get("cache_key")
+    last_refresh = response.get("last_refresh")
+    if isinstance(cache_key, str) and isinstance(last_refresh, str):
+        update_cached_state(  # Updating the relevant InsightCachingState
+            insight.team_id,
+            cache_key,
+            last_refresh,
+            result=None,  # Not caching the result here, since in HogQL this is the query runner's responsibility
+        )
```

Comment:
The only behavioral change in this PR: `TestDashboard::test_refresh_cache` revealed we weren't updating `InsightCachingState` in here. This change brings us fully back on track with tracking which insights need a background refresh